pkg/utils/event.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package utils
+
+import "sync"
+
+// Event is a synchronization primitive that lets an arbitrary number
+// of concurrent waiters to wait for an event to happen 1 or more times.
+type Event struct {
```

Comment:
We have `sync2` for synchronization primitives. Any reason you are not using `sync.Cond`?

This should definitely be a separate PR. And this needs tests.